plugins/needs_restarting.py

```diff
@@ -40,6 +40,50 @@
 NEED_REBOOT = ['kernel', 'glibc', 'linux-firmware', 'systemd', 'dbus',
                'dbus-broker', 'dbus-daemon']
 
+def get_options_from_dir(filepath):
+    """
+    Provide filepath as string if single dir or list of strings
+    Return set of package names contained in files under filepath
+    """
+    base = dnf.Base()
+    base.fill_sack()
+    packages = set()
+    replaced = set()
+    configfiles = set()
+    options = set()
+
+    # find dirs
+    if isinstance(filepath, str):
+        replaced = { filepath } if os.path.isdir(filepath) else set()
```

Comment:
you could call it by `get_options_from_dir(["path1", "path2"])`

I can remove it if you want

Reply:
Yes, but it is not used atm. and it's questionable whether it ever will be needed (I'd say likely it won't). I'd keep it simple for now, it can be extended very easily if it ever is needed.